### ISSUE TITLE
!Sub with Mapping

### ISSUE THREAD

**Issue by martysweet** [yyolk/cloudformation-js-yaml-schema#1]
Currently, cloudformation-js-yaml-schema expects !Sub to be of a Scalar type, expecting a string. The following YAML would be invalid within the Schema.

```
Name: !Sub
  - www.${Domain}
  - { Domain: !Ref RootDomainName }
```

According to http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-sub.html the !Sub function should support type String (for `!Sub ${something}` and a Sequence to allow for mapping functionality.

I will create a pull request with the !Sub type in the Sequence tags array and with accompanying unit tests. This will allow for both formats of the !Sub shorthand.

**Comment by yyolk** [comment]
Thanks @martysweet, I missed this completely since I'm usually using `!Sub` like so: 

```yaml
Name: !Sub ${DnsPrefix}.${DnsZone}
```

I will merge your fix shortly :)

**Comment by yyolk** [comment]
closed with #2
